A workflow to auto bump versions

Rishit-dagli opened the pull request:
**What this PR does / why we need it**:

Updating versions on behalf of the contributor since we already have a run that checks the version would be very helpful as discussed in yesterday's meeting and in #500 . With this PR being merged we should expect the contributing workflow to look like this:

- Contributor makes a PR not necessarily running the `version.sh` file themselves
- Reviewer decides if it needs to be a patch, minor or major
- Reviewer comments one of of the following the PR (these comments have no effects if made on an issue):
    - /version major
    - /version minor
    - /version patch
- GitHub Actions makes the required changes and adds a commit to the same PR 
- GitHub Actions adds the appropriate label, one of:
   - version/major
   - version/minor
   - version/patch 

**Special notes for your reviewer**:

Before merging this PR we need to create labels: `version/minor`, `version/major` and `version/patch`

Closes #500 

**If applicable**:
- [ ] this PR has an associated PR with documentation in [akri-docs](https://github.com/project-akri/akri-docs) (WIP, this also requires changes to some documentation in this repo which are made in #511)
- [ ] this PR contains unit tests
- [x] added code adheres to standard Rust formatting (`cargo fmt`)
- [x] code builds properly (`cargo build`)
- [x] code is free of common mistakes (`cargo clippy`)
- [x] all Akri tests succeed (`cargo test`)
- [x] inline documentation builds (`cargo doc`)
- [x] version has been updated appropriately (`./version.sh`)
- [x] all commits pass the [DCO bot check](https://probot.github.io/apps/dco/) by being signed off -- see the failing DCO check for instructions on how to retroactively sign commits

Rishit-dagli commented:
Here is an end to end example of this workflow in action: https://github.com/Rishit-dagli/akri-500/pull/2

bfjelds commented:
love it!!

bfjelds commented on .github/workflows/update-versions.yml:
```diff
@@ -0,0 +1,62 @@
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
```
does it make a difference whether the PR is from a fork or local when this runs? (assume it is fine, but want to make sure the fork PRs work)

Rishit-dagli replied:
Nice catch. I seem to have found the mistake I was doing her though let me run a quick tests and push some changes

Rishit-dagli commented:
@bfjelds I addressed the comments with the latest commits

Feel free to see a sample run of this workflow at https://github.com/Rishit-dagli/akri-500/pull/4

romoh commented:
Thanks for doing that change @Rishit-dagli !
